controllers/routingctrl/fixtures_test.go

```diff
@@ -46,12 +47,65 @@ func addRoutingRequirementsToSvc(ctx context.Context, exportedSvc *corev1.Servic
 	Expect(errExportSvc).ToNot(HaveOccurred())
 }
 
-func createComponentRequiringPlatformRouting(ctx context.Context, componentName, mode, appNs string) (*unstructured.Unstructured, error) {
+// createComponentRequiringPlatformRouting creates a new component with the specified routing modes.
+func createComponentRequiringPlatformRouting(ctx context.Context, componentName, appNs string, isExternal, isPublic bool) (*unstructured.Unstructured, error) {
 	component, errCreate := test.CreateUnstructured(componentResource(componentName, appNs))
 	Expect(errCreate).ToNot(HaveOccurred())
 
-	// set component's "routing.opendatahub.io/export-mode" annotation to the specified mode.
-	metadata.ApplyMetaOptions(component, annotations.RoutingExportMode(mode))
+	if isExternal {
+		metadata.ApplyMetaOptions(component, annotations.ExternalMode())
+	}
+
+	if isPublic {
+		metadata.ApplyMetaOptions(component, annotations.PublicMode())
+	}
 
 	return component, envTest.Client.Create(ctx, component)
 }
+
+type exportModeAction struct {
+	mode  string
+	value string
+}
+
+var (
+	enablePublic           = exportModeAction{mode: annotations.PublicMode().Key(), value: "true"}
+	disableExternal        = exportModeAction{mode: annotations.ExternalMode().Key(), value: "false"}
+	removeExternal         = exportModeAction{mode: annotations.ExternalMode().Key(), value: ""}
+	removePublic           = exportModeAction{mode: annotations.PublicMode().Key(), value: ""}
+	enableNonSupportedMode = exportModeAction{
```

Comment:
For this comment, you specifically mean:
`	enableNonSupportedMode = exportModeAction{
`
right? IE instead of checking for no public/external resources check that we correctly verify the annotation to be  a non supported one.

Reply:
We can check that we didn't create those resources, but ultimately I wonder what kind of behaviour we really want to ensure for such a case. I think it's beyond this PR, maybe add `TODO` in this test to cover it when we figure out reporting/events.